codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -782,11 +783,53 @@ picToObj' pic =
             return $ (unsafeCoerce arr :: JSVal)
     setProps xs obj = liftIO $ void $ mapM (\(s, v) -> setProp s v obj) xs
 
+-- x is the maximum display length (Int)
+-- y is the input string we want to trim
+-- if string (y) is less than x, then we return y
+-- otherwise, replace the interior of string with ...
+
+trim :: Int -> String -> String
+trim x y = case (x > (length y)) of
```

Comment:
Also, `case (x > (length y)) of True -> ...; False` is the same as an `if` expression:

```haskell
trim x y = if x > length y then y else ...
```